From ff65246a3a97322c319cb2580198194f4dfbdbda Mon Sep 17 00:00:00 2001
From: Khem Raj <khem.raj@oss.qualcomm.com>
Date: Wed, 8 Jul 2026 00:00:00 +0000
Subject: [PATCH] libfmt 12.2.0 support: use fmt/format.h not fmt/core.h

Since libfmt 12.2.0, <fmt/core.h> is a shim that only includes
<fmt/base.h>, which does not declare fmt::format(). Building against it
fails with:

  src/net/ToString.cxx:80:16: error: no member named 'format' in namespace 'fmt'
  src/lib/avahi/Publisher.cxx:37:9: error: no member named 'format' in namespace 'fmt'

Include <fmt/format.h> directly in the files that call fmt::format().
This mirrors the upstream fix (MPD commits 88d5b5248d71 and 109c26258b9d)
extended to the files present in this release that call fmt::format().

Upstream-Status: Backport [https://github.com/MusicPlayerDaemon/MPD/commit/88d5b5248d71]
Signed-off-by: Khem Raj <khem.raj@oss.qualcomm.com>
---
 src/client/New.cxx                             | 2 +-
 src/db/plugins/upnp/UpnpDatabasePlugin.cxx     | 2 +-
 src/db/update/Playlist.cxx                     | 2 +-
 src/input/plugins/AlsaInputPlugin.cxx          | 2 +-
 src/input/plugins/QobuzClient.cxx              | 2 +-
 src/input/plugins/QobuzLoginRequest.cxx        | 2 +-
 src/input/plugins/QobuzTrackRequest.cxx        | 2 +-
 src/lib/avahi/Publisher.cxx                    | 2 +-
 src/lib/nfs/FileReader.cxx                     | 2 +-
 src/net/ToString.cxx                           | 2 +-
 src/output/Init.cxx                            | 2 +-
 src/output/plugins/httpd/HttpdClient.cxx       | 2 +-
 src/output/plugins/httpd/IcyMetaDataServer.cxx | 2 +-
 src/song/AddedSinceSongFilter.cxx              | 2 +-
 src/song/AudioFormatSongFilter.cxx             | 2 +-
 src/song/ModifiedSinceSongFilter.cxx           | 2 +-
 src/storage/plugins/NfsStorage.cxx             | 2 +-
 17 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/src/client/New.cxx b/src/client/New.cxx
index 84788b5fd..65d08b907 100644
--- a/src/client/New.cxx
+++ b/src/client/New.cxx
@@ -17,7 +17,7 @@
 #include "Log.hxx"
 #include "Version.h"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 
diff --git a/src/db/plugins/upnp/UpnpDatabasePlugin.cxx b/src/db/plugins/upnp/UpnpDatabasePlugin.cxx
index d45b242f5..52d8e48b9 100644
--- a/src/db/plugins/upnp/UpnpDatabasePlugin.cxx
+++ b/src/db/plugins/upnp/UpnpDatabasePlugin.cxx
@@ -24,7 +24,7 @@
 #include "util/StringSplit.hxx"
 #include "config/Block.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 #include <string>
diff --git a/src/db/update/Playlist.cxx b/src/db/update/Playlist.cxx
index bc0215e43..d6db77d9c 100644
--- a/src/db/update/Playlist.cxx
+++ b/src/db/update/Playlist.cxx
@@ -20,7 +20,7 @@
 #include "fs/Traits.hxx"
 #include "Log.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 inline void
 UpdateWalk::UpdatePlaylistFile(Directory &directory,
diff --git a/src/input/plugins/AlsaInputPlugin.cxx b/src/input/plugins/AlsaInputPlugin.cxx
index f8099686c..9cc5b4ac9 100644
--- a/src/input/plugins/AlsaInputPlugin.cxx
+++ b/src/input/plugins/AlsaInputPlugin.cxx
@@ -27,7 +27,7 @@
 
 #include <alsa/asoundlib.h>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 
diff --git a/src/input/plugins/QobuzClient.cxx b/src/input/plugins/QobuzClient.cxx
index efa184f81..4e710ffbe 100644
--- a/src/input/plugins/QobuzClient.cxx
+++ b/src/input/plugins/QobuzClient.cxx
@@ -4,7 +4,7 @@
 #include "QobuzClient.hxx"
 #include "lib/crypto/MD5.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 #include <stdexcept>
diff --git a/src/input/plugins/QobuzLoginRequest.cxx b/src/input/plugins/QobuzLoginRequest.cxx
index 41d4dcabf..7e2e6b64c 100644
--- a/src/input/plugins/QobuzLoginRequest.cxx
+++ b/src/input/plugins/QobuzLoginRequest.cxx
@@ -6,7 +6,7 @@
 #include "QobuzSession.hxx"
 #include "lib/curl/Form.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <nlohmann/json.hpp>
 
 #include <cassert>
diff --git a/src/input/plugins/QobuzTrackRequest.cxx b/src/input/plugins/QobuzTrackRequest.cxx
index 59b6cf480..3de5e7ea0 100644
--- a/src/input/plugins/QobuzTrackRequest.cxx
+++ b/src/input/plugins/QobuzTrackRequest.cxx
@@ -5,7 +5,7 @@
 #include "QobuzErrorParser.hxx"
 #include "QobuzClient.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <nlohmann/json.hpp>
 
 using std::string_view_literals::operator""sv;
diff --git a/src/lib/avahi/Publisher.cxx b/src/lib/avahi/Publisher.cxx
index 1ac4d61e6..174f57198 100644
--- a/src/lib/avahi/Publisher.cxx
+++ b/src/lib/avahi/Publisher.cxx
@@ -13,7 +13,7 @@
 #include <avahi-common/malloc.h>
 #include <avahi-common/alternative.h>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 
diff --git a/src/lib/nfs/FileReader.cxx b/src/lib/nfs/FileReader.cxx
index bc14fd325..805babef6 100644
--- a/src/lib/nfs/FileReader.cxx
+++ b/src/lib/nfs/FileReader.cxx
@@ -10,7 +10,7 @@
 
 #include <nfsc/libnfs.h> // for struct nfs_stat_64
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 #include <cstring>
diff --git a/src/net/ToString.cxx b/src/net/ToString.cxx
index 6f9706c0a..539ad5c61 100644
--- a/src/net/ToString.cxx
+++ b/src/net/ToString.cxx
@@ -6,7 +6,7 @@
 #include "IPv4Address.hxx"
 #include "net/Features.hxx" // for HAVE_TCP, HAVE_IPV6, HAVE_UN
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <algorithm>
 #include <cassert>
diff --git a/src/output/Init.cxx b/src/output/Init.cxx
index 9c76321a6..4df70e9eb 100644
--- a/src/output/Init.cxx
+++ b/src/output/Init.cxx
@@ -24,7 +24,7 @@
 #include "util/StringAPI.hxx"
 #include "Log.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 #include <stdexcept>
diff --git a/src/output/plugins/httpd/HttpdClient.cxx b/src/output/plugins/httpd/HttpdClient.cxx
index b097cd38f..73ed2af55 100644
--- a/src/output/plugins/httpd/HttpdClient.cxx
+++ b/src/output/plugins/httpd/HttpdClient.cxx
@@ -13,7 +13,7 @@
 #include "util/StringSplit.hxx"
 #include "Log.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 
diff --git a/src/output/plugins/httpd/IcyMetaDataServer.cxx b/src/output/plugins/httpd/IcyMetaDataServer.cxx
index 0314089f1..0796bffcb 100644
--- a/src/output/plugins/httpd/IcyMetaDataServer.cxx
+++ b/src/output/plugins/httpd/IcyMetaDataServer.cxx
@@ -5,7 +5,7 @@
 #include "tag/Tag.hxx"
 #include "util/TruncateString.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <iterator>
 
diff --git a/src/song/AddedSinceSongFilter.cxx b/src/song/AddedSinceSongFilter.cxx
index efc394fd0..dfe397542 100644
--- a/src/song/AddedSinceSongFilter.cxx
+++ b/src/song/AddedSinceSongFilter.cxx
@@ -6,7 +6,7 @@
 #include "time/ISO8601.hxx"
 #include "util/StringBuffer.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/song/AudioFormatSongFilter.cxx b/src/song/AudioFormatSongFilter.cxx
index 69bd2b1a8..c1f209730 100644
--- a/src/song/AudioFormatSongFilter.cxx
+++ b/src/song/AudioFormatSongFilter.cxx
@@ -5,7 +5,7 @@
 #include "LightSong.hxx"
 #include "util/StringBuffer.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/song/ModifiedSinceSongFilter.cxx b/src/song/ModifiedSinceSongFilter.cxx
index 43879c789..715f65e15 100644
--- a/src/song/ModifiedSinceSongFilter.cxx
+++ b/src/song/ModifiedSinceSongFilter.cxx
@@ -6,7 +6,7 @@
 #include "time/ISO8601.hxx"
 #include "util/StringBuffer.hxx"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/storage/plugins/NfsStorage.cxx b/src/storage/plugins/NfsStorage.cxx
index 27f72e02b..fe092e041 100644
--- a/src/storage/plugins/NfsStorage.cxx
+++ b/src/storage/plugins/NfsStorage.cxx
@@ -28,7 +28,7 @@ extern "C" {
 #include <nfsc/libnfs-raw-nfs.h>
 }
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 #include <string>
